modules/text2vec-openai/vectorizer/objects.go

```diff
@@ -87,3 +115,181 @@ func (v *Vectorizer) getVectorizationConfig(cfg moduletools.ClassConfig) ent.Vec
 		Dimensions:   settings.Dimensions(),
 	}
 }
+
+// batchWorker is a go routine that handles the communication with the vectorizer
+//
+// On the high level it has the following steps:
+//  1. It receives a batch job
+//  2. It splits the job into smaller vectorizer-batches if the token limit is reached. Note that objects from different
+//     batches are not mixed with each other to simplify returning the vectors.
+//  3. It sends the smaller batches to the vectorizer
+func (v *Vectorizer) batchWorker() {
+	rateLimit := &ent.RateLimits{}
+	texts := make([]string, 0, 100)
+	firstRequest := true
+	timePerToken := 0.0
+	batchTookInS := float64(0)
+BatchLoop:
+	for job := range v.jobQueueCh {
+		// the total batch should not take longer than 60s to avoid timeouts. We will only use 40s here to be safe
+		batchStart := time.Now()
+
+		objCounter := 0
+		vecBatchOffset := 0 // index of first object in current vectorizer-batch
+		tokensInCurrentBatch := 0
+		texts = texts[:0]
+
+		conf := v.getVectorizationConfig(job.cfg)
+
+		// we don't know the current rate limits without a request => send a small one
+		for firstRequest {
+			var err error
+			rateLimit, err = v.makeRequest(job, job.texts[:1], conf, vecBatchOffset)
+			if err != nil {
+				for j := 0; j < len(job.texts); j++ {
+					job.errs[j] = err
```

Comment:
should skipObjects be assigned with errors?

(applies also to other assignments)

Reply:
I set skipObjects if an object should not be vectorized. For example a user could have already provided a vector for some objects for some named vectors but not for others or an object needs only be revectorized for some named vectors (because of different input properties).

To avoid creating a new []object list for each vectorizer, I added the skip-list. This also simplifies the mapping of the returned vectors

Reply:
I will add more comments to usecases/modules/vectorizer.go to make it clearer

Reply:
I understand why skip list was added. For sure it helps processing batch and avoid creating new slices for each vectorizer.
My concern was rather errors being assigned to every remaining object in the batch, even if it is expected to be skipped. Aren't they supposed to be omitted here and error assigned only to those which needed vectorization?

Reply:
ah yes - don't think it matter, but still fixed